### PR TITLE
Add events.

### PR DESCRIPTION
**Pull request by aranajhonny**
* `issueOpened`
* `pullRequestOpened`
* `PullRequestCollection`
* `PullRequestPage`
* `PullRequest`

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -372,10 +376,43 @@
           {
             "name": "watchersCount",
             "type": "Int"
+          },
+          {
+            "name": "pullRequest",
```
Comment (juancampa) [inline config.json]: pluralize?

File: config.json [hunk 2]
```diff
@@ -372,10 +376,43 @@
...
+            "name": "pullRequestOpened",
+            "params": [
+              {
+                "name": "issue",
+                "type": "Issue"
+              },
+              {
+                "name": "pullRequest",
+                "type": "PullRequest"
+              }
```
Comment (juancampa) [inline config.json]: These two should be refs

File: config.json [hunk 3]
```diff
@@ -523,7 +560,256 @@
...
+        "computedFields": [
+          {
+            "name": "one",
+            "params": [
+              {
+                "name": "id",
+                "type": "String"
+              }
+            ],
+            "type": "String"
```
Comment (juancampa) [inline config.json]: Shouldn't the type here be "Hook"?

File: config.json [hunk 4]
```diff
@@ -523,7 +560,256 @@
...
+            "name": "createdAt",
+            "params": [],
+            "type": "String"
+          }
+        ],
+        "actions": [],
+        "events": []
+      },
+      {
+        "name": "Events",
```
Comment (juancampa) [inline config.json:733]: Empty type

File: index.js [hunk 5]
```diff
@@ -134,6 +173,7 @@ export const Repository = {
   subscribersCount({ source }) { return source['subscribers_count']; },
   networkCount({ source }) { return source['network_count']; },
   issues({ self, source }) { return {}; },
+  pullRequest({ self, source }) { return {}; },
```
Comment (juancampa) [inline index.js]: feel free to remove unused params

File: index.js [hunk 6]
```diff
@@ -165,3 +205,107 @@ export const Issue = {
...
+        // dispatch Event
+        const repoRef = root.users.one({ name: owner }).repos.one({ name: repo })
+        await repoRef.pullRequestOpened.dispatch({
+            issue: repo.issues.one({ number: issueNumber }),
+            pullRequest: repo.pullRequest.one({ number: issueNumber })
+        });
+      }
+    }
+  const timer = Number.parseInt(result.meta['x-poll-interval']);
+  await program.setTimer(`${owner}/${repo}`, timer);
```
Comment (juancampa) [inline index.js]: minor: can use `key` here.

File: index.js [hunk 7]
```diff
@@ -165,3 +205,107 @@ export const Issue = {
...
+    return {
+      items: res.data,
+      next: getNextPageRef(self.page(args), res),
+    };
+  }
+}
+
+export const Hook = {
+  self({ self, parent, source }) {
+    return self || parent.ref.pop().pop().push('one', { id: source.hook_id });
```
Comment (juancampa) [inline index.js]: `self` resolvers should now be written in this form:

`return parent.parent.parent.one({ id: source.hook_id })`

For clarity, here's what it means:

 - `parent` is a ref to the `[Hook]` inside a page
 - `parent.parent` is a ref to the the `HookPage` in the collection
 - `parent.parent.parent` is a ref to the `HookCollection`
 - And from the ref to the collection we just go into the `one` field with the right id

**Review by juancampa — CHANGES_REQUESTED** [review]

**Comment by aranajhonny** [comment]
@juancampa  check https://github.com/aranajhonny/subscribe-to-pr

File: index.js [hunk 8]
```diff
@@ -110,10 +120,39 @@ export const RepositoryCollection = {
...
 export const Repository = {
   self({ self, parent, source }) {
     return self || parent.ref.pop().pop().push('one', { name: source.name });
   },
+  issueOpened: {
+    async subscribe({ self }) {
+      const { name: owner } = self.match(root.users.one);
+      const { name: repo } = self.match(root.users.one.repos.one);
+      
+      await program.setTimer(`${owner}/${repo}`, 0, 10);
```
Comment (juancampa) [inline index.js:133]: This should trigger the timer immediately instead of 10s in the future. Call `timer(`${owner}/${repo}`)` directly?

**Review by juancampa — CHANGES_REQUESTED** [review]

**Comment by aranajhonny** [comment]
prueba

**Pull request merged**